exec/java-exec/src/test/java/org/apache/drill/test/OperatorTestBuilder.java

```diff
@@ -0,0 +1,300 @@
+        Assert.assertEquals((int) expectedTotalRowsOpt.orElse(0), actualTotalRows);
+        // We are done, we don't have any expected result to compare
+        return;
+      }
+
+      if (combineOutputBatches) {
+        final RowSet expectedBatch = expectedResults.get(0);
+        final RowSet actualBatch = DirectRowSet.fromSchema(
+          physicalOpUnitTestBase.operatorFixture.allocator, expectedBatch.schema());
+        final VectorContainer actualBatchContainer = actualBatch.container();
+        actualBatchContainer.setRecordCount(0);
+
+        final int numColumns = expectedBatch.schema().size();
+        List<MutableInt> totalBytesPerColumn = new ArrayList<>();
+
+        for (int columnIndex = 0; columnIndex < numColumns; columnIndex++) {
+          totalBytesPerColumn.add(new MutableInt());
+        }
+
+        // Get column sizes for each result batch
+
+        final List<List<RecordBatchSizer.ColumnSize>> columnSizesPerBatch = actualResults.stream().map(rowSet -> {
+          switch (rowSet.indirectionType()) {
+            case NONE:
+              return new RecordBatchSizer(rowSet.container()).columnsList();
+            case TWO_BYTE:
+              return new RecordBatchSizer(rowSet.container(), ((RowSet.SingleRowSet) rowSet).getSv2()).columnsList();
+            default:
+              throw new UnsupportedOperationException();
+          }
+        }).collect(Collectors.toList());
+
+        // Get total bytes per column
+
+        for (List<RecordBatchSizer.ColumnSize> columnSizes: columnSizesPerBatch) {
+          for (int columnIndex = 0; columnIndex < numColumns; columnIndex++) {
+            final MutableInt totalBytes = totalBytesPerColumn.get(columnIndex);
+            final RecordBatchSizer.ColumnSize columnSize = columnSizes.get(columnIndex);
+            totalBytes.add(columnSize.getTotalDataSize());
+          }
+        }
+
+        for (int columnIndex = 0; columnIndex < numColumns; columnIndex++) {
+          final ValueVector valueVector = actualBatchContainer
+            .getValueVector(columnIndex)
+            .getValueVector();
+
+          if (valueVector instanceof FixedWidthVector) {
+            ((FixedWidthVector) valueVector).allocateNew(actualTotalRows);
+          } else if (valueVector instanceof VariableWidthVector) {
+            final MutableInt totalBytes = totalBytesPerColumn.get(columnIndex);
+            ((VariableWidthVector) valueVector).allocateNew(totalBytes.getValue(), actualTotalRows);
+          } else {
+            throw new UnsupportedOperationException();
+          }
+        }
+
+        try {
+          int currentIndex = 0;
+
+          for (RowSet actualRowSet: actualResults) {
+            final Copier copier;
+            final VectorContainer rowSetContainer = actualRowSet.container();
+            rowSetContainer.setRecordCount(actualRowSet.rowCount());
+
+            switch (actualRowSet.indirectionType()) {
+              case NONE:
```

Comment:
Why? The code doesn't support output batches with a selection vector yet. I'd rather add support for it if it's actually needed.

Reply:
But you are adding corresponding RowSet inside actualResults [here](https://github.com/apache/drill/pull/1344/files/332b5f34d43f8f546824619c966b978eb5788b47#diff-de4ffc656d08e47bd1b8c716dca26f3eR109) based on output container has SV or not.

Reply:
Yeah I guess I did (forgot it was so long ago). I don't remember the details, but I added the code you linked to because it was easy to add. When implementing the copy here I ran into an issue (I don't quite remember what, it's been a couple months). Since handling sv2s wasn't required for testing HashAgg I gave up on it decided to leave it unimplemented. 

I'd rather not pursue completing the code to handle sv2s in this PR. The intention here was to test HashAgg, if somebody needs to handle output batches with selection vectors for another operator, they can add it as a separate PR. I'll remove the code snippet you've referenced and throw an UnsupportedException there as well. I made a jira to track this feature https://issues.apache.org/jira/browse/DRILL-6698 and will reference the jira number in the code.